### PR TITLE
Windows: Poll loop to check if mount is truly done

### PR DESCRIPTION
**Pull request by infeo**
Several times in the past CI tests on Windows failed due for the mount not being ready, although `fuse_init()` was already called.

jfuse uses https://github.com/winfsp/winfsp on Windows and according to its owner, there is no proper way of checking at what point the filesystem is accessible from the outside (see https://github.com/winfsp/winfsp/discussions/440). It was suggested to poll the volume until it successfully returns, but query the volume itself on the one hand requires using the native `CreateFileW` function and on the other admin rights.

The now implemented solution blocks the mount method until the fuse filesystem recieves a `getAttr(...)` call to a specific, hardcoded path.

### CODE REVIEW

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java [hunk 1]
```diff
@@ -41,8 +51,29 @@ public void mount(String progName, Path mountPoint, String... flags) throws Moun
...
+	private void waitForMountingToComplete(Path mountPoint) throws InterruptedException {
+		var probe = Files.getFileAttributeView(mountPoint.resolve(MOUNT_PROBE.substring(1)), BasicFileAttributeView.class);
+		while (!accessProbeSuccess.get()) {
+			try {
+				probe.readAttributes();
+				Files.readAttributes(mountPoint, BasicFileAttributes.class);
+			} catch (IOException e) {
+				//do nothing
+			} finally {
+				Thread.sleep(333);
```
Comment (overheadhunter) [inline jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:71]: If `getattr` is called and `accessProbeSuccess` is set to true, we will still sleep one last time. Which is not necessary.

If we replaced the `AtomicBoolean` with a `CountDownLatch`, we could instead use a do-while loop with `probeSucceeded.await(333, TimeUnit.MILLIS)` as the loop's condition.

**Review by overheadhunter — CHANGES_REQUESTED** [review]
I think I found a more elegant means for synchronization, which kills two birds with one stone.

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java (hunk 1 above)
Comment (overheadhunter) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:71]: which also reads like prosa:

> do read attributes while awaiting the probe

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java [hunk 2]
```diff
@@ -139,6 +170,10 @@ private void destroy(MemoryAddress addr) {
 	@VisibleForTesting
 	int getattr(MemoryAddress path, MemoryAddress stat, MemoryAddress fi) {
 		try (var scope = MemorySession.openConfined()) {
+			var pathObj = path.getUtf8String(0);
```
[the remark this answers is not in the document]
Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:169]: Implemented in https://github.com/cryptomator/jfuse/pull/9/commits/b33dabc0cad8020b1a14d6ae3c3e71d57f0858f5

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java (hunk 1 above)
Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java]: Implemented in https://github.com/cryptomator/jfuse/pull/9/commits/b33dabc0cad8020b1a14d6ae3c3e71d57f0858f5

**Review by overheadhunter — APPROVED** [review]

**Comment by overheadhunter** [comment]
I just noticed that we're waiting indefinitely, if the file system doesn't [support](https://github.com/cryptomator/jfuse/blob/66827353f9e351b8cfe0cbb958da61f8227bcedc/jfuse-api/src/main/java/org/cryptomator/jfuse/api/FuseOperations.java#L47-L50) `getattr`.

We should rather always set `fuse_operations.getattr`, as we have [previously done it with init](https://github.com/cryptomator/jfuse/blob/0.1.0/jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java#L28-L50).

**Pull request merged**